src/models/network-v2.ts

```diff
@@ -417,8 +497,16 @@ export class Network_v2 extends Model<Network_v2Methods> implements Deployable {
     return this.callTx(this.contract.methods.cidBountyId(cid));
   }
 
-  async getDelegationOf(address: string) {
-    return this.callTx(this.contract.methods.getDelegationsFor(address));
+  async getDelegationsOf(address: string): Promise<Delegation[]> {
+    const delegations = await this.callTx(this.contract.methods.getDelegationsFor(address));
+    const mappedDelegations = delegations.map((delegation, index) => ({
+      ...delegation,
+      id: index,
+      amount: fromSmartContractDecimals(+delegation.amount, this.settlerToken.decimals)
+    }));
+    const higherThanZero = mappedDelegations.filter(delegation => delegation.amount > 0);
+
+    return higherThanZero;
```

Comment:
`higherThanZero` is redundant, you can return the filtered `delegations` map result, 

```ts
return delegations.map(delegation).filter(({amount}) => +amount);
```

*note: `delegation` is an exported `src/util` that return `({...delegation, id: index, amount....`